tests/system/src/test/java/test/robot/javafx/stage/FocusParentWindowOnChildCloseTest.java

```diff
@@ -0,0 +1,159 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.robot.javafx.stage;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Scene;
+import javafx.scene.control.Alert;
+import javafx.scene.control.Button;
+import javafx.scene.control.ButtonType;
+import javafx.scene.input.MouseButton;
+import javafx.scene.layout.StackPane;
+import javafx.scene.robot.Robot;
+import javafx.stage.Stage;
+import javafx.stage.WindowEvent;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import test.util.Util;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+// See JDK8210973
+public class FocusParentWindowOnChildCloseTest {
+    static Robot robot;
+    static Button button;
+    static Stage stage;
+    static Stage stage2;
+    static Scene scene;
+    static Alert alert;
+    static CountDownLatch startupLatch = new CountDownLatch(2);
+    static CountDownLatch alertShownLatch;
+    static CountDownLatch alertCloseLatch;
+
+    @Test(timeout = 15000)
+    public void focusRightParentOnChildWindowClose() throws Exception {
+        Thread.sleep(400);
+        clickButton();
+        fireOkInAlert();
+        Assert.assertTrue("Stage 1 should be focused", stage.isFocused());
+    }
+
+    private void clickButton() throws Exception {
+        alertShownLatch = new CountDownLatch(1);
+        mouseClick(button.getLayoutX() + button.getWidth() / 2, button.getLayoutY() + button.getHeight() / 2);
+
+        Thread.sleep(400);
+        waitForLatch(alertShownLatch, 10, "Failed to show Alert");
+    }
+
+    @BeforeClass
+    public static void initFX() throws Exception {
+        new Thread(() -> Application.launch(TestApp.class, (String[]) null)).start();
+        waitForLatch(startupLatch, 10, "FX runtime failed to start.");
+    }
+
+    @AfterClass
+    public static void exit() {
+        Platform.runLater(() -> {
+            stage.hide();
+            stage2.hide();
+        });
+        Platform.exit();
+    }
+
+    private void mouseClick(double x, double y) {
+        Util.runAndWait(() -> {
+            robot.mouseMove((int) (scene.getWindow().getX() + scene.getX() + x),
+                    (int) (scene.getWindow().getY() + scene.getY() + y));
+            robot.mousePress(MouseButton.PRIMARY);
+            robot.mouseRelease(MouseButton.PRIMARY);
+        });
+    }
+
+    private static void fireOkInAlert() throws Exception {
+        alertCloseLatch = new CountDownLatch(1);
+        Platform.runLater(() -> {
+                    Button okButton = (Button) alert.getDialogPane().lookupButton(ButtonType.OK);
+                    System.out.println(okButton);
```

Comment:
The print statement is not required here. please remove.

Reply:
Oops, forgot that.